include/cetl/variable_length_array.hpp

```diff
@@ -1226,6 +1226,54 @@ class VariableLengthArray : protected VariableLengthArrayBase<T, Allocator>
 
 #endif
 
+    /// Returns a reference to the first element in the array.
+    /// Calling this method on an empty array is undefined.
+    /// @return Reference to the first element in the array.
+    constexpr reference front()
+    {
+        CETL_DEBUG_ASSERT(size() > 0, "CDE_vla_004: Calling front() on an empty array is undefined.");
+        return data_[0];
+    }
+
+    /// Returns a const reference to the first element in the array.
+    /// Calling this method on an empty array is undefined.
+    /// @return Constant reference to the first element in the array.
+    constexpr const_reference front() const
+    {
+        CETL_DEBUG_ASSERT(size() > 0, "CDE_vla_005: Calling front() on an empty array is undefined.");
+        return data_[0];
+    }
+
+    /// Returns a reference to the last element in the array.
+    /// Calling this method on an empty array is undefined.
+    /// @return Reference to the last element in the array.
+    constexpr reference back()
+    {
+        // a, perhaps naive, attempt to constrain undefined behaviour considering
```

Comment:
Why would you want to constrain undefined behavior?

Reply:
It's a crap shoot to be sure but, if you are debugging a misuse of this method it'll be easier to find the problem in your disassembly if the segfault was really close to this method instead of off in the greater weeds. Frankly, this isn't a tested property but it's all undefined behaviour so I could just as well have pointed the reference to a full-text copy of James Joyce's Ulysses and I wouldn't be violating the interface contract (why does CETL add so much bloat to my firmware!? It's like they embedded War and Peace in 4-byte unicode into their library...)